### PR TITLE
Process push notifications. RTDSDK-3082

### PR DESCRIPTION
**Pull request by Pavel-Cherentsov**

### CODE REVIEW

File: src/components/firebase-support.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    },
+    function (err) {
+      console.log("ServiceWorker registration failed:", err);
+    }
+  );
+};
+
+export const subscribeFcmNotifications = (convoClient: Client) => {
+  Notification.requestPermission()
+    .then(() => {
```
Comment (sveta-leonovets-twi) [inline src/components/firebase-support.ts:20]: `async await` js syntax'd look more readable in this case

**Review by sveta-leonovets-twi — APPROVED** [review]

File: src/components/firebase-support.ts (hunk 1 above)
Comment (berkus) [reply src/components/firebase-support.ts:20]: Makes sense to use async/await everywhere, also annotate async functions as such.

File: src/components/firebase-support.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import firebase from "firebase/compat/app";
+import "firebase/compat/messaging";
+
+export const initFcmServiceWorker = async () => {
+  firebase.initializeApp((window as any).firebaseConfig);
+
+  navigator.serviceWorker.register("firebase-messaging-sw.js").then(
```
Comment (InfiniteRain) [inline src/components/firebase-support.ts:8]: use async/await syntax in here

File: src/components/firebase-support.ts [hunk 3]
```diff
@@ -39,3 +39,19 @@ export const subscribeFcmNotifications = (convoClient: Client) => {
...
+
+export const showNotification = (pushNotification: PushNotification) => {
+  const notificationTitle = "Notification: " + pushNotification.type;
+  const notificationOptions = {
+    body: pushNotification.body,
+    icon: "favicon.ico",
+  };
+
+  const notification = new Notification(notificationTitle, notificationOptions);
+  notification.onclick = function (event) {
```
Comment (InfiniteRain) [inline src/components/firebase-support.ts:51]: use arrow function syntax to keep things consistent: `notification.onclick = (event) => {`

**Review by InfiniteRain — CHANGES_REQUESTED** [review]
I see a lot of places where `.then` is being used in place of `await`. We should use the `async/await` syntax everywhere where we use asynchronous functions.

File: src/components/firebase-support.ts (hunk 1 above)
Comment (Pavel-Cherentsov) [reply src/components/firebase-support.ts]: Fixed

File: src/components/firebase-support.ts (hunk 2 above)
Comment (Pavel-Cherentsov) [reply src/components/firebase-support.ts]: Fixed

File: src/components/firebase-support.ts (hunk 3 above)
Comment (Pavel-Cherentsov) [reply src/components/firebase-support.ts]: Fixed

**Review by InfiniteRain — APPROVED** [review]

File: public/firebase-messaging-sw.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+importScripts('https://www.gstatic.com/firebasejs/9.6.3/firebase-messaging.js');
+importScripts('firebase-config.js');
+
+if (firebaseConfig) {
+  firebase.initializeApp(firebaseConfig);
+  console.log(`${new Date().toJSON()}  [firebase-messaging-sw] Initialized messaging`);
+
+  firebase.messaging().setBackgroundMessageHandler(payload => {
+    console.log(`${new Date().toJSON()}  [firebase-messaging-sw] Received background message`, payload);
+    const notificationTitle = payload.data.twi_message_type;
```
Comment (aleksandrsivanovs) [inline public/firebase-messaging-sw.js:11]: here we should use static method Client.parsePushNotification(<pushdata>) and then access it through returned object of type PushNotification

File: public/firebase-messaging-sw.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    const notificationOptions = {
+      body: payload.data.twi_body,
```
Comment (aleksandrsivanovs) [inline public/firebase-messaging-sw.js:13]: here we should use static method Client.parsePushNotification(<pushdata>) and then access it through returned object of type PushNotification

**Pull request merged**